include/vcpkg/base/system.process.h

```diff
@@ -80,9 +80,31 @@ namespace vcpkg
 
     Path get_exe_path_of_current_process();
 
-    struct ExitCodeAndOutput
+    struct ProcessResult
     {
-        int exit_code;
+        ProcessResult() : ProcessResult("not executed") { }
+        ProcessResult(int exit_code) : exit_code(exit_code) { }
+        ProcessResult(std::string api_error) : api_error(std::move(api_error)) { }
```

Comment:
Is there a good reason this type isn't just `ExpectedL<int>`?

Reply:
> This should probably be `LocalizedString`?

yes, but first I wanted to get a design approval :)

> Is there a good reason this type isn't just ExpectedL<int>?

Yes, it resulted in a huge mess in https://github.com/microsoft/vcpkg-tool/pull/482/files

Reply:
That doesn't look like a huge mess to me? :shrug:

Reply:
It was only 15% done and I don't like that you always have to write something like this
```c++
if (run_nuget_commandline(cmd).has_error_or([](int exit_code) { return exit_code != 0; }))
```
when you want to check the exit code instead of 
```c++
if (!run_nuget_commandline(cmd).successful())
// or even 
if (!run_nuget_commandline(cmd))
```

Reply:
It should be:

```
if (run_nuget_commandline(cmd).value_or_exit(VCPKG_LINE_INFO))
```

not anything that needs a lambda or something like that.

Reply:
Or we could even do:

```
bool process_run_successful(const ExpectedL<int>& ex) {
    if (auto i = ex.get()) {
        return *i == 0;
    }
    return false;
}
```

I just see gradually designing another thing that looks like Optional or Expected here and that scares me.

Reply:
I understand your concerns. But imo this is only a small helper so that we don't have to change the over 100 places where `cmd_execute` is called.